addons/product/models/product_attribute.py

```diff
@@ -133,14 +137,21 @@ def _unlink_except_used_on_product(self):
 
     # === ACTION METHODS === #
 
+    def action_archive(self):
+        if self.number_related_products:
+            raise UserError(_(
+                "You cannot archive this record as there are still products linked to this record",
+            ))
```

Comment:
`action_archive` can be called on multiple records at once, you should loop on `self` here